llvm/lib/CodeGen/SelectionDAG/LegalizeDAG.cpp

```diff
@@ -1496,11 +1500,12 @@ SDValue SelectionDAGLegalize::ExpandInsertToVectorThroughStack(SDValue Op) {
     Ch = DAG.getTruncStore(
         Ch, dl, Part, SubStackPtr,
         MachinePointerInfo::getUnknownStack(DAG.getMachineFunction()),
-        VecVT.getVectorElementType());
+        VecVT.getVectorElementType(), ElementAlignment);
```

Comment:
I'm still confused about what is going on with the alignment. Can you assert the resulting StoreSDNode's getAlign is the element alignment (the actual element alignment, not the misnamed ElementAlignment here)?

Reply:
I have added the assert when PartVT is not a vector. If I have to add the same assert when PartVT is a vector, should I compare the alignment of the StoreSDNode with the alignment of the PartVT or with the alignment of an element of the vector?
Also, I hope I am calculating the element alignment correctly. Please let me know if I am not doing so.

Reply:
Done